src/pages/OfferPage.js

```diff
@@ -0,0 +1,61 @@
+import React, { useEffect, useContext, useState } from "react";
+import { useParams, Redirect } from "react-router-dom";
+
+import OfferContent from "../components/HomePage/SearchResultsArea/Offer/OfferContent";
+import Offer from "../components/HomePage/SearchResultsArea/Offer/Offer";
+import { getOffer } from "../services/getOfferService";
+import { CardContent } from "@material-ui/core";
+
+export const OfferPageControllerContext = React.createContext();
+
+export const OfferPageController = () => {
+    const { id } = useParams();
+    const [offer, setOffer] = useState(null);
+    const [error, setError] = useState(null);
+    const [loading, setLoading] = useState(true);
+
+    useEffect(() => {
+        getOffer(id)
+            .then((offer) => {
+                setOffer(new Offer(offer));
+                setLoading(false);
+            })
+            .catch(() => {
+                setError("UnexpectedError");
+            });
+        return () => {
+        };
+    }, [id]);
```

Comment:
Just to make it clearer, this is where the SWR for offers would come in handy. You could create a wrapper hook that called the API through SWR, and you would have a mutate and revalidate functions available for when you want to change the offers (disable, hide etc), which you would pass down through the controller's context (simply return them below.

Then, when the inner components wanted to change the offers (in this case, the AdminReasonModal I think), they can call the functions to mutate locally, and call for a revalidation after the request to hide/disable etc (which will trigger swr to just re-fetch the offer from the API, which should now have the correct values in the fields for isHidden, hiddenReason etc

Be aware that the OfferContent cannot rely on using the context to access these functions, since it is being used on the search widget as well, which does not provide this context. Instead, make the offerContent receive the functions as props, and pass them here, and an alternative function in the search results widget case (which won't be using SWR, but you can still mutate the offers which are stored in the Redux store locally, I believe)